### DISCUSSION TITLE
Where is source code for create_plot_disk() so I can hack NVME temp control in 'create plot'

### DISCUSSION THREAD

**Discussion by room101-dev** [ericaltendorf/plotman#513] [General]
I'm trying to put in emergency failure prediction into plot-manager, but I need the internals of the plot-creation process.

The chia-network seems to be hiding the source for this process, without this source its impossible to follow temperature

I would prefer to throttle down the usage on NVME from inside the code, but it appears there is NO code.

Does anybody know where this code resides? Sure asking question here because we need things done, and if you go on chia-network they'll blow you away, as you all know.

Ask a problem about plot crashes and they'll say 'memory problem', of course sure you could say memory in NVME problem, but really the problem is in the NVME controllers, nada about 'ddr ram' on mobo

chia/plotting/create_plots.py make a call to create_plot_disk() ... but where is the source ?? So I can hack it up, and stop 'create plot' from killing my NVME drives???

I assuming here that CHIA is a 'black-box' if that is the case, then we don't know if really what they're doing to our plot drives :(

If this is really 'open source' then where is the source?

**Comment by calarndt** [comment]
check the header files maybe???
I haven't looked at any of the code but that is where I would start...

**Reply by room101-dev** [reply]
Got it, it was in chia-pos marked as 'html' my gawd they make stuff difficult here, already hacked it up, now it shuts down if my NVME's over-heat :)

The wrappers for python use different names than the C++ calls, ... the project says HTML only I knew I was looking for C++, somebody here pointed me to the direction. All is done now, I have what I want.

**Reply by lnb2g** [reply]
would you add a copy function to replace rsync as an archive option for us? rysnc is nice but its too slow.

**Comment by altendky** [comment]
https://github.com/Chia-Network/chia-blockchain/search?q=create_plot_disk&type=
https://github.com/Chia-Network/chia-blockchain/blob/9cc908678b1255c9a520c322302ba35084676e08/chia/plotting/create_plots.py#L164
https://github.com/chia-network/chiapos
https://github.com/Chia-Network/chiapos/search?q=create_plot_disk
https://github.com/Chia-Network/chiapos/blob/ce6ccc28ea300bfbad7e054a3f30433eb6090cf5/python-bindings/chiapos.cpp#L39
https://github.com/Chia-Network/chiapos/blob/ce6ccc28ea300bfbad7e054a3f30433eb6090cf5/src/plotter_disk.hpp#L59

Maybe be a bit more understanding.
